### ISSUE TITLE
Log contains  pg_temp functions

### ISSUE THREAD

**Issue by Elena-alexeenko** [sequelize/sequelize#14266]
When using Model.findOrCreate to create a a Model the values are being entered to the log as they are,
I came to realization that this is what causes it : 
` 
   valueQuery = `${tmpTable}INSERT${replacements.ignoreDuplicates} INTO ${quotedTable} (${replacements.attributes})${replacements.output} VALUES (${replacements.values})${onDuplicateKeyUpdate}${replacements.onConflictDoNothing}${valueQuery}`;
    emptyQuery = `${tmpTable}INSERT${replacements.ignoreDuplicates} INTO ${quotedTable}${replacements.output}${onDuplicateKeyUpdate}${replacements.onConflictDoNothing}${emptyQuery}`;

if (this._dialect.supports.EXCEPTION && options.exception) {
      const dropFunction = 'DROP FUNCTION IF EXISTS pg_temp.testfunc()';

      if (returningModelAttributes.length === 0) {
        returningModelAttributes.push('*');
      }

      const delimiter = `$func_${uuidv4().replace(/-/g, '')}$`;
      const selectQuery = `SELECT (testfunc.response).${returningModelAttributes.join(', (testfunc.response).')}, testfunc.sequelize_caught_exception FROM pg_temp.testfunc();`;

      options.exception = 'WHEN unique_violation THEN GET STACKED DIAGNOSTICS sequelize_caught_exception = PG_EXCEPTION_DETAIL;';
      valueQuery = `CREATE OR REPLACE FUNCTION pg_temp.testfunc(OUT response ${quotedTable}, OUT sequelize_caught_exception text) RETURNS RECORD AS ${delimiter} BEGIN ${valueQuery} RETURNING * INTO response; EXCEPTION ${options.exception} END ${delimiter} LANGUAGE plpgsql; ${selectQuery} ${dropFunction}`;
    } 
`

The entry to the log is: 
CREATE
OR REPLACE FUNCTION pg_temp.testfunc(
    OUT response "user",
    OUT sequelize_caught_exception text
) RETURNS RECORD AS $ func_92a3e0fdeaee4ec2aff6dd8d44c1ec4d $ BEGIN
INSERT INTO
    "user" (
        "email",
        "password",
        "createdAt",
        "updatedAt"
    )
VALUES
    (
        'abcd123@gmail.com',
        '131072$4bc44b3a01fdc82848a5b7a31e6adf8c63cf26aef6d15102b1d778ceba2033119333def8c5d656df643680feec659fc1c0f69de2e29a4bbc201bfb2a2e2571b5$2b50b5d1b39b7209b12d6a06d38b5f9b5d509a5dfbfcfefab71b76a7ae0e0a5aad549f2bb90c118043d29d85724392cee5d4d072e2488b4b24d6b7efc5b33c40',
        '2022-03-22 13:03:13.148 +00:00',
        '2022-03-22 13:03:13.148 +00:00'
    ) RETURNING * INTO response;

EXCEPTION
WHEN unique_violation THEN GET STACKED DIAGNOSTICS sequelize_caught_exception = PG_EXCEPTION_DETAIL;

END $ func_92a3e0fdeaee4ec2aff6dd8d44c1ec4d $ LANGUAGE plpgsql;

SELECT
    (testfunc.response)."email",
    (testfunc.response)."password",
    (testfunc.response)."createdAt",
    (testfunc.response)."updatedAt",
    testfunc.sequelize_caught_exception
FROM
    pg_temp.testfunc();

DROP FUNCTION IF EXISTS pg_temp.testfunc();

In the comment to code it is written that this is mostly of internal use.

- Sequelize version: 6.6.5
- Node.js version: 14.15.5
- If TypeScript related: TypeScript version:4.3.5
- Database & Version: postgreSQL 11
- Connector library : pg

**Comment by ephys** [comment]
I'm not sure I understand the issue. If you have logging enabled, sequelize will log all queries it sends to the database.

**Comment by Elena-alexeenko** [comment]
The issue is that only in findOrCreate the values of the query are being displayed in the log. In other queries this exception is not reached or logged

**Comment by ephys** [comment]
I think that's because the other methods use bind parameters, but this one doesn't

**Comment by soryy708** [comment]
That's right, other queries are build parameterized so when logged the values are `$1`, `$2`...
But looks like `findOrCreate` is a special case, violating the implicit interface in an unexpected way.

**Comment by soryy708** [comment]
Looks like in this case Elena's problem is that `findOrCreate` surprises the user in that it logs a hashed (and salted?) password, instead of the usual behavior of logging a parameterized query.
To prevent this from happening, I think an ad-hoc workaround can be constructed by using `findOne`, inspecting it's returned values, and conditionally using `create`. To keep things ACID, wrap it in a transaction.
This is clearly less efficient, because it requires more roundtrips to the database, and a more performant solution would involve an altered `findOrCreate`.
